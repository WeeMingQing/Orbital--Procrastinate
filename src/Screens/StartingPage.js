import React, {useRef, useEffect} from "react";
import {View, Animated, TouchableOpacity, Text,  StyleSheet,Image} from "react-native"







export default StartPage = ({navigation}) =>{
    const opacity = useRef(new Animated.Value(0)).current
    useEffect(()=>{
        fadeIn();
        setTimeout(()=>{
            fadeOut();
        },500);
        setTimeout(()=>{navigation.replace("Login Page")},1100)
    },[])
    function fadeIn(){
        Animated.timing(opacity,{
            toValue: 1,
            duration: 500,
            useNativeDriver:true
        }).start()
    };

    function fadeOut(){
        Animated.timing(opacity,{
            toValue: 0.2,
            duration: 600,
            useNativeDriver:true
        }).start()
    };



    return (
        <View style = {styles.body}>
            <Image
            style={styles.image}
            resizeMode = "stretch"
            source={require("../../assets/!Procrastinate_Logo.png")}
          />
            <Animated.View style = {[{width:"100%", opacity,alignSelf:"center"}]}>
                <Text style = {[styles.text2, "font-family: Cursive"]}>From</Text>
                <Text style = {[styles.text, "font-family: fantasy"]}>!Procrastinate</Text>

            </Animated.View>


        </View>
    )
};

const styles = StyleSheet.create({
    body: {
        flex: 1,
        backgroundColor: "#ffffff",
        justifyContent: "center",
        padding: 10,
    },
    image: {
        height:350,
        width: 350,
        marginTop:80,
        marginRight:15,
        alignItems: "center",
        justifyContent: "center",
        borderRadius:2,
      },
    text: {
      alignSelf: "center",
      justifyContent: "center",
      color: "#21d6ff",
      fontSize: 25,
      fontStyle: "italic",
    },
    text2: {
      alignSelf: "center",
      justifyContent: "center",
      color: "black",
      fontSize: 20,
      fontStyle: "italic",
    },
  });