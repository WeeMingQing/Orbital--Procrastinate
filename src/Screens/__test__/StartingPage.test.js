import React from "react";
import renderer from 'react-test-renderer'
import StartingPage from "../StartingPage";

// pass
it("test Starting page",  () => {
  const tree = renderer.create(<StartingPage/>).toJSON();
  expect(tree).toMatchSnapshot()  

})
