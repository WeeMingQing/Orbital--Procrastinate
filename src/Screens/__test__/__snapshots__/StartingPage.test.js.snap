// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Starting page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../assets/!Procrastinate_Logo.png",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 2,
        "height": 350,
        "justifyContent": "center",
        "marginRight": 15,
        "marginTop": 80,
        "width": 350,
      }
    }
  />
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "center",
        "opacity": 0,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "black",
            "fontSize": 20,
            "fontStyle": "italic",
            "justifyContent": "center",
          },
          "font-family: Cursive",
        ]
      }
    >
      From
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#21d6ff",
            "fontSize": 25,
            "fontStyle": "italic",
            "justifyContent": "center",
          },
          "font-family: fantasy",
        ]
      }
    >
      !Procrastinate
    </Text>
  </View>
</View>
`;
